### ISSUE TITLE
Customizable Cores

### ISSUE THREAD

**Issue by LaserFont0103** [ChaosInitiative/Portal-2-Community-Edition#575]
## Which component should be improved?
Entity

## Describe your suggestion
There should be an way to create custom cores.

> The core should be able to have custom animations, textures, and sounds.
> when you spawn a core you can choose from preset cores (like wheatley, curiosity core, etc.) or you can import your own assets from your pc or from a link.
> There should also be an option to link the cores to triggers and allow the cores to act according to certain actions

## Expected result
In the end, the cores should be able to work very similarly to the cores in game and shouldn't require any more modding besides Portal 2: Community Edition

**Comment by Luke18033** [comment]
I'm not exactly sure what's being suggested here? Most of this stuff can already be done in the base game. Custom animations and textures can already be implemented by recompiling the model, and custom voicelines can be added with the choreo system. You can also have triggers that make cores play animations using choreo scenes or `scripted_sequence`s.

It seems like you might be suggesting this in the context of a Puzzlemaker. P2CE doesn't currently have a puzzlemaker and won't for a long time, but even when one is added, it's called a **puzzle**maker, not a storymaker. You should still use Hammer if you want to have a story with characters like cores.

**Comment by slambo312** [comment]
> I'm not exactly sure what's being suggested here? Most of this stuff can already be done in the base game. Custom animations and textures can already be implemented by recompiling the model, and custom voicelines can be added with the choreo system. You can also have triggers that make cores play animations using choreo scenes or `scripted_sequence`s.
> 
> It seems like you might be suggesting this in the context of a Puzzlemaker. P2CE doesn't currently have a puzzlemaker and won't for a long time, but even when one is added, it's called a **puzzle**maker, not a storymaker. You should still use Hammer if you want to have a story with characters like cores.

i believe he means an easier way of making cores, for less experienced users

**Comment by Luke18033** [comment]
I still don't see a reason why that needs to be a part of P2CE, someone could make an external tool to quickly make custom cores just as easily.

**Comment by LaserFont0103** [comment]
> > I'm not exactly sure what's being suggested here? Most of this stuff can already be done in the base game. Custom animations and textures can already be implemented by recompiling the model, and custom voicelines can be added with the choreo system. You can also have triggers that make cores play animations using choreo scenes or `scripted_sequence`s.
> > It seems like you might be suggesting this in the context of a Puzzlemaker. P2CE doesn't currently have a puzzlemaker and won't for a long time, but even when one is added, it's called a **puzzle**maker, not a storymaker. You should still use Hammer if you want to have a story with characters like cores.
> 
> i believe he means an easier way of making cores, for less experienced users

that's what I meant, I want it to be easier to customize cores without the need for external tools, but of course the feature can be held off until a puzzle maker is being worked on.

**Comment by Luke18033** [comment]
So it is for a Puzzlemaker then? In that case it's not going to be implemented, the puzzlemaker isn't for stories and that won't change in P2CE.

**Comment by DeliWasTaken** [comment]
I don't believe he meant for puzzlemaker, I just think he meant it's a tool that he can wait for after puzzlemaker is finished. Though, I don't think making cores is really that difficult of a task in the first place.

**Comment by Luke18033** [comment]
In that case, again, **how does this have to do with P2CE?** Take Allison's custom portal gun maker, someone could easily make something like that but for custom cores, it doesn't require code access at all. The only thing P2CE will add is the ability to use a separate model for a custom core instead of overriding the default one, but if that's what this is suggesting then it's a duplicate of #130.

**Comment by DeliWasTaken** [comment]
Yes, I fully agree.
